docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,182 @@
+# Storage node installation and update on Linux
+
+## Abstract
+
+The design document outlines the steps needed to ease installation and provide auto-update of storage nodes on Linux without the use of Docker.
+
+
+## Background
+
+
+The idea is to have two main components, like for Windows: the storage node binary and a storage node updater that will take care of updating the storage node binary according to the rollout version mechanism.
+
+The parts we need:
+- storagenode as a service
+- a system for updating the storagenode binary aka. the updater (with rollout versioning support)
+- a system for updating the updater 
+- a way to collect the configuration data from the user during the installation
+- packaging to ship the above
+
+Linux systems are not unified as windows is. Hence, we have to make choices that will allow us to cover most linux users, with special regards to the raspbian system.
+
+Most commonly used linux distributions seem to be (there are no reliable stats):
+- debian-based (Debian, Ubuntu, Mint, Kali, Raspbian)
+- Red Hat-based (Fedora)
+- Arch Linux (Manjaro is on the rise according to [distrowatch](https://distrowatch.com/dwres.php?resource=popularity)).
+
+All these distributions are shipped with systemD as a service manager.
+
+
+
+
+## Design
+The installer will be a debian package. We choose to auto-update the binary, even if this will make the package not following debian guidelines
+- When installing the package, it should prompt for user input for:
+  - Wallet address
+  - Email
+  - External address/port
+  - Advertised bandwidth
+  - Advertised storage
+  - Identity directory
+  - Storage directory
```

Comment:
there could probably be good defaults for many of these

Reply:
There is https://wiki.archlinux.org/index.php/XDG_Base_Directory

Reply:
From what I'm reading, the XDG base directory values have to be defined by the user, so I guess most of the time they will be empty. I'm using manjaro and these values are not defined, except XDG_RUNTIME_DIR set by pam_systemd.

Reply:
They have exact default values what you should use when they are not defined.

`XDG_DATA_HOME`:
* Should default to `$HOME/.local/share`